espresso/streamer.go

```diff
@@ -299,3 +312,18 @@ func (s *EspressoStreamer[B]) HasNext(ctx context.Context) bool {
 
 	return false
 }
+
+// This function allows to "pin" the Espresso block height corresponding to the last safe batch
+func (s *EspressoStreamer[B]) confirmEspressoBlockHeight(safeL1Origin eth.BlockID) error {
+	hotshotState, err := s.EspressoLightClient.
+		FinalizedState(&bind.CallOpts{BlockNumber: new(big.Int).SetUint64(safeL1Origin.Number)})
+	if errors.Is(err, bind.ErrNoCode) {
```

Comment:
Why not return an error in this case as well? In practice setting the `fallbackHotShotPos` to 0 will mean the batcher to be inactive for a very long time.

Reply:
This is to handle a case where contract is not yet deployed at this L1 height, which we can interpret as hotshot position corresponding to this L1 height is 0